File: app/src/main/java/com/concrete/android/challenge/ui/splash/SplashActivity.java
package com.concrete.android.challenge.ui.splash;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import com.concrete.android.challenge.BR;
import com.concrete.android.challenge.R;
import com.concrete.android.challenge.databinding.ActivitySplashBinding;
import com.concrete.android.challenge.ui.base.BaseActivity;
import com.concrete.android.challenge.ui.main.MainActivity;

/**
 * @author Thiago Corredo
 * @since 2019-07-12
 */
public class SplashActivity extends BaseActivity<ActivitySplashBinding, SplashViewModel>
    implements SplashNavigator {

  private ActivitySplashBinding activitySplashBinding;
  private SplashViewModel splashViewModel;

  @Override public SplashViewModel getViewModel() {
    splashViewModel = ViewModelProviders.of(this, factory).get(SplashViewModel.class);
    return splashViewModel;
  }

  @Override public int getBindingVariable() {
    return BR.viewModel;
  }

  @Override public int getLayoutId() {
    return R.layout.activity_splash;
  }

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    activitySplashBinding = getViewDataBinding();
    splashViewModel.setNavigator(this);

    Typeface type = Typeface.createFromAsset(getAssets(), "fonts/fff_tusj.ttf");
    activitySplashBinding.title.setTypeface(type);
  }

  @Override public void openMainActivity() {
    Intent intent = MainActivity.newIntent(this);
    startActivity(intent);
  }
}

File: app/src/main/java/com/rafael/fernandes/desafioconcrete/ui/activities/splash/SplashActivity.kt
package com.rafael.fernandes.desafioconcrete.ui.activities.splash

import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import com.rafael.fernandes.desafioconcrete.R
import com.rafael.fernandes.desafioconcrete.databinding.SplashActivityBinding
import com.rafael.fernandes.desafioconcrete.ui.activities.main.MainActivity
import com.rafael.fernandes.desafioconcrete.ui.base.BaseActivity

class SplashActivity : BaseActivity<SplashActivityBinding, SplashViewModel>() {

    override fun layoutId(): Int = R.layout.splash_activity

    override fun beforeSetContentView() {
        window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
    }

    override fun myOnCreate(savedInstanceState: Bundle?) {
        mViewModel.getLiveData().observe(this, Observer {
            gotoNextScreen(MainActivity::class.java)
        })
    }
}